### ISSUE TITLE
Error in not accepting JWST x1d NIRISS data

### ISSUE THREAD

**Issue by Aseman7** [astropy/specutils#1004]
Hello, when I want to use Specutils for NIRSpec data it works correctly using its x1d.fits file. But it does not recognize the same x1d.fits file extracted for NIRISS data. My Python version is 3.9.12 and Specutils version is 1.9.0. Where do you think the problem is? 

f = fits.open('jw02589003001_04101_00001-seg003_nis_x1d.fits')  
jwst_spec = Spectrum1D.read(f)

I got this error:

IORegistryError: Format could not be identified based on the file name or contents, please provide a 'format' argument.
The available formats are:
      Format      Read Write Auto-identify
----------------- ---- ----- -------------
      6dFGS-split  Yes    No           Yes
    6dFGS-tabular  Yes    No           Yes
    APOGEE apStar  Yes    No           Yes
   APOGEE apVisit  Yes    No           Yes
APOGEE aspcapStar  Yes    No           Yes
            ASCII  Yes    No           Yes
             ECSV  Yes    No           Yes
          HST/COS  Yes    No           Yes
         HST/STIS  Yes    No           Yes
             IPAC  Yes    No           Yes
         JWST c1d  Yes    No           Yes
         JWST s2d  Yes    No           Yes
         JWST s3d  Yes    No           Yes
         JWST x1d  Yes    No           Yes
      MUSCLES SED  Yes    No           Yes
       MaNGA cube  Yes    No           Yes
        MaNGA rss  Yes    No           Yes
     SDSS spPlate  Yes    No           Yes
 SDSS-I/II spSpec  Yes    No           Yes
 SDSS-III/IV spec  Yes    No           Yes
 Subaru-pfsObject  Yes    No           Yes
             iraf  Yes    No           Yes
     tabular-fits  Yes   Yes           Yes
       wcs1d-fits  Yes   Yes           Yes

 

Thanks a lot!

**Comment by rosteen** [comment]
Hi, I took a quick look on MAST and saw NIRISS `x1dints.fits` files for this observation, but not a `xid.fits` file - is this a file/spectrum that you created yourself, or am I just missing it on MAST?

Either way, what happens if you add the `format` argument to the `read` command, i.e. `jwst_spec = Spectrum1D.read(f, format="JWST x1d")`?

**Comment by Aseman7** [comment]
Thanks a lot for your reply. 

Yes, although apparently, I have extracted a x1d file using the JWST pipeline tool. But the fact is that what I got is a x1dints file, not a x1d file for NIRISS SOSS data, so Specutils can't read it.

Specutils doesn't have a problem with its extracted x1d files of NIRSpec data and works correctly. But as for NIRISS SOSS data:

I searched and found that the NIRISS SOSS data is a time series-type observation. And for time series data, the only supported results are the time series-based results, or the 'ints' files. 

So I have to convert x1dints to x1d from another way. "x1dints" has some EXTRACT1D HDUs, since each EXTRACT1D HDU contains the spectrum at each integration. I have to average the individual integrations for extracting x1d files for NIRISS SOSS data. One of the specialist said that there is no specific JWST calibration code that does that directly. And he said that I have to do this in another way, for example using JWST.datamodels.

I am trying to do that :), thanks.
